### PR TITLE
Add gradient filter

### PR DESCRIPTION
**Pull request by tfarago**

### CODE REVIEW

File: src/kernels/gradient.cl [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    if (idy == height - 1) {
+        dy = 0;
+    } else if (idy == 0) {
+        dy = 2;
+    }
+
+    return 0.5f * (input[(idy + dy) * width + idx] - input[(idy + dy - 2) * width + idx]);
+}
+
+kernel void DIRECTION_HORIZONTAL (global float *input,
```
Comment (tfarago) [inline src/kernels/gradient.cl]: I know, but it's just so convenient...

File: src/kernels/gradient.cl [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                          int idy,
+                          int width) {
+    int dx = 1;
+    if (idx == width - 1) {
+        dx = 0;
+    } else if (idx == 0) {
+        dx = 2;
+    }
+
+    return 0.5f * (input[idy * width + idx + dx] - input[idy * width + idx + dx - 2]);
```
Comment (tfarago) [inline src/kernels/gradient.cl]: BTW this is how [numpy](https://docs.scipy.org/doc/numpy/reference/generated/numpy.gradient.html) does it.

**Comment by tfarago** [comment]
The `both_abs` will be very useful for computing the tomographic metric.

File: src/kernels/gradient.cl (hunk 1 above)
Comment (matze) [reply src/kernels/gradient.cl]: Yeah, looks ugly but I get your point.

**Comment by tfarago** [comment]
Ready to merge.

**Pull request merged**